### PR TITLE
added breadcrumbs component

### PR DESCRIPTION
**Pull request by darkares23**
## Purpose/Description:

This PR introduces enhancements to the Breadcrumbs component, expanding its capabilities to support a variety of child components such as text, links, buttons, and icons. These changes aim to improve the flexibility and user experience of the component.

### Screenshots:
Captura de pantalla 2023-10-10 a la(s) 11 46 47 a m

## Jira Link:

Link to the relevant Jira issue or task. (e.g., [BRAN-154](https://collagegroup.atlassian.net/browse/BRAN-154?atlOrigin=eyJpIjoiZWFmMzNiMWJmZTc1NGNhMWE5NjZkOTIzYTA3NzBjNzMiLCJwIjoiaiJ9))

## Test Plan:

Test Scenario 1: Ensure basic breadcrumbs display correctly.

Steps:
1. Navigate to the component's Storybook page.
2. Access the "Basic" story.
3. Observe the displayed breadcrumbs.
Expected Result: 
Breadcrumbs should be displayed in a linear fashion, with clear legible text.

Test Scenario 2: Verify icon support in breadcrumbs.
Steps:
1. Navigate to the component's Storybook page.
2. Access the "With Icons" story.
3. Observe the displayed breadcrumbs.
Expected Result: Breadcrumbs items should include icons where specified, aligned properly with the text.

### Reviewer Instructions:

- Focus on the responsiveness of the breadcrumbs, especially when icons are included.
- Check the compatibility of the new changes with existing usages of the Breadcrumbs component in the application.
- Validate the appearance in both light and dark modes.

[BRAN-154]: https://collagegroup.atlassian.net/browse/BRAN-154?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/atoms/Breadcrumbs/Breadcrumbs.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import HomeIcon from '@mui/icons-material/Home';
```
Comment (diego20) [inline src/components/atoms/Breadcrumbs/Breadcrumbs.stories.tsx]: Have my doubts about this. I think we should use only phosphor icons. Pending confirmation from Vernon

File: src/components/atoms/Breadcrumbs/Breadcrumbs.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import Link from '@mui/material/Link';
+import { Meta, StoryFn } from '@storybook/react';
+import { Box, Breadcrumbs, BreadcrumbsProps, Button } from 'src/components';
+
+export default {
+  title: 'Components/Breadcrumbs',
+  component: Breadcrumbs,
+} as Meta;
+
+const Template: StoryFn<BreadcrumbsProps> = (args) => <Breadcrumbs {...args} />;
```
Comment (diego20) [inline src/components/atoms/Breadcrumbs/Breadcrumbs.stories.tsx]: Suggestion: Emphasize more the presence of children
```suggestion
const Template: StoryFn<typeof Breadcrumbs> = ({ children, ...args }) => (
  <Breadcrumbs {...args}>{children}</Breadcrumbs>
);
```

File: src/components/atoms/Breadcrumbs/Breadcrumbs.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
+import HomeIcon from '@mui/icons-material/Home';
+import Link from '@mui/material/Link';
+import { Meta, StoryFn } from '@storybook/react';
+import { Box, Breadcrumbs, BreadcrumbsProps, Button } from 'src/components';
+
+export default {
```
Comment (diego20) [inline src/components/atoms/Breadcrumbs/Breadcrumbs.stories.tsx:27]: Add `argTypes` to display the correct types of all the props you want to expose from the component.
I believe this way we won't expose unnecessary props such as `itemsAfterCollapse`, `ref`, `slots`, `classes`, etc. And also display the correct UI according to the types of props we want to leave like: `maxItems` and `separator`... I don't see any other prop we would care to display. Any opinions @Alderr? `sx`? I don't think `children` We can't actually do anything by modifying its values in the UI

File: src/components/atoms/Breadcrumbs/Breadcrumbs.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    <Button key="7" variant="link">
+      Item
+    </Button>,
+    <Button key="8" variant="link">
+      Item 2
+    </Button>,
+  ],
+};
+
+export const WithButtons = Template.bind({});
```
Comment (diego20) [inline src/components/atoms/Breadcrumbs/Breadcrumbs.stories.tsx]: I don't see much usefulness in this variant. It's purpose I think is to show what we could render in the breadcrumb but I think that is achieved in all other variants and this in particular is a version we will never use. I suggest removing this.

Comment (diego20) [reply src/components/atoms/Breadcrumbs/Breadcrumbs.stories.tsx]: I would instead apply a variant with a custom selector

File: src/components/atoms/Breadcrumbs/Breadcrumbs.stories.tsx (hunk 3 above)
Comment (Alderr) [reply src/components/atoms/Breadcrumbs/Breadcrumbs.stories.tsx:27]: +1

File: src/components/atoms/Breadcrumbs/Breadcrumbs.stories.tsx (hunk 1 above)
Comment (Alderr) [reply src/components/atoms/Breadcrumbs/Breadcrumbs.stories.tsx]: +1 if mockups don't **require** us to use them, we should only use phosphor

File: src/components/atoms/Breadcrumbs/Breadcrumbs.tsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import NavigateNextIcon from '@mui/icons-material/NavigateNext';
+import MuiBreadcrumbs, {
+  BreadcrumbsProps as MuiBreadcrumbsProps,
+} from '@mui/material/Breadcrumbs';
+
+export type BreadcrumbsProps = MuiBreadcrumbsProps;
+
+export const Breadcrumbs: React.FC<BreadcrumbsProps> = ({
```
Comment (Alderr) [inline src/components/atoms/Breadcrumbs/Breadcrumbs.tsx]: remove `React.FC`, i like the way we do it in the dashboard

File: src/components/atoms/Breadcrumbs/Breadcrumbs.tsx [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import NavigateNextIcon from '@mui/icons-material/NavigateNext';
+import MuiBreadcrumbs, {
+  BreadcrumbsProps as MuiBreadcrumbsProps,
+} from '@mui/material/Breadcrumbs';
+
+export type BreadcrumbsProps = MuiBreadcrumbsProps;
```
Comment (Alderr) [inline src/components/atoms/Breadcrumbs/Breadcrumbs.tsx]: I'd remove this and just use `MuiBreadcrumpsProps`, we should also do this for other components

File: src/components/index.ts [hunk 7]
```diff
@@ -7,3 +7,6 @@ export type { ButtonProps } from './atoms/Button';
 
 export { Box } from './atoms/Box';
 export type { BoxProps } from './atoms/Box';
+
```
Comment (Alderr) [inline src/components/index.ts:10]: 💯

File: src/components/atoms/Breadcrumbs/index.ts [hunk 8]
```diff
@@ -0,0 +1,2 @@
+export { Breadcrumbs } from './Breadcrumbs';
```
Comment (Alderr) [inline src/components/atoms/Breadcrumbs/index.ts:1]: 💯

**Review by Alderr — COMMENTED** [review]
LGTM, needs a couple changes and that's it

**Review by diego20 — APPROVED** [review]
LGTM

**Pull request merged**